toqito/helper/bell_notation_conversions.py

```diff
@@ -0,0 +1,748 @@
+            ax_val = fc_mat[1 + x, 0]
+            for y in range(ib):
+                by_val = fc_mat[0, 1 + y]
+                axby_val = fc_mat[1 + x, 1 + y]
+                # V[0,0,x,y] = P(++,xy) = (1 + <Ax> + <By> + <AxBy>)/4
+                v_mat[0, 0, x, y] = 1 + ax_val + by_val + axby_val
+                # V[0,1,x,y] = P(+-,xy) = (1 + <Ax> - <By> - <AxBy>)/4
+                v_mat[0, 1, x, y] = 1 + ax_val - by_val - axby_val
+                # V[1,0,x,y] = P(-+,xy) = (1 - <Ax> + <By> - <AxBy>)/4
+                v_mat[1, 0, x, y] = 1 - ax_val + by_val - axby_val
+                # V[1,1,x,y] = P(--,xy) = (1 - <Ax> - <By> + <AxBy>)/4
+                v_mat[1, 1, x, y] = 1 - ax_val - by_val + axby_val
+        v_mat = v_mat / 4
+
+    return v_mat
+
+
+def fp_to_cg(v_mat: np.ndarray, behaviour: bool = False) -> np.ndarray:
+    r"""Convert a Bell functional or behaviour from Full Probability (FP) to Collins-Gisin (CG) notation.
+
+    The Full Probability (FP) notation represents the full probability distribution
+    :math:`V(a, b, x, y) = P(a, b | x, y)`, where :math:`a` (0 to oa-1), :math:`b` (0 to ob-1) are
+    outcomes and :math:`x` (0 to ia-1), :math:`y` (0 to ib-1) are inputs. It's stored as a 4D
+    numpy array `V[a, b, x, y]`. The Collins-Gisin (CG) notation for a Bell functional or
+    behaviour is represented by a matrix (see :func:`cg_to_fc`).
+
+    This function converts from FP to CG notation.
+
+    Examples
+    ==========
+
+    Consider the CHSH inequality functional in FP notation:
+    (Here V represents coefficients, not probabilities)
+
+    >>> import numpy as np
+    >>> from toqito.helper import fp_to_cg
+    >>> chsh_fp = np.zeros((2, 2, 2, 2))
+    >>> chsh_fp[0, 0, 0, 0] = 1
+    >>> chsh_fp[0, 0, 0, 1] = -1
+    >>> chsh_fp[0, 0, 1, 0] = -1
+    >>> chsh_fp[0, 0, 1, 1] = 1
+    >>> fp_to_cg(chsh_fp)
+    array([[ 0.,  0.,  0.],
+           [ 0.,  1., -1.],
+           [ 0., -1.,  1.]])
+
+    Consider a behaviour (probability distribution) in FP notation (standard PR box):
+
+    >>> pr_box = np.zeros((2, 2, 2, 2))
+    >>> pr_box[0, 0, 0, 0] = 0.5 # p(0,0|0,0)
+    >>> pr_box[1, 1, 0, 0] = 0.5 # p(1,1|0,0)
+    >>> pr_box[0, 0, 0, 1] = 0.5 # p(0,0|0,1)
+    >>> pr_box[1, 1, 0, 1] = 0.5 # p(1,1|0,1)
+    >>> pr_box[0, 0, 1, 0] = 0.5 # p(0,0|1,0)
+    >>> pr_box[1, 1, 1, 0] = 0.5 # p(1,1|1,0)
+    >>> pr_box[0, 1, 1, 1] = 0.5 # p(0,1|1,1)
+    >>> pr_box[1, 0, 1, 1] = 0.5 # p(1,0|1,1)
+    >>> fp_to_cg(pr_box, behaviour=True) # doctest: +NORMALIZE_WHITESPACE
+    array([[1. , 0.5, 0.5],
+           [0.5, 0.5, 0.5],
+           [0.5, 0.5, 0. ]])
+
+    :param v_mat: The probability tensor V[a, b, x, y] in Full Probability notation.
+    :param behaviour: If True, assume input is a behaviour (default: False, assume functional).
+    :return: The matrix in Collins-Gisin notation.
+
+    .. note::
+        This function is adapted from the QETLAB MATLAB package function `FP2CG`.
```

Comment:
```suggestion
        This function is adapted from the QETLAB MATLAB package function ``FP2CG``.
```
You have to use double ` because `rst` renders single ` in italics.